news/5788.bugfix

```diff
@@ -0,0 +1 @@
+Make requirements' messages more consistent.
```

Comment:
If this PR is doing more than just changing how the messages look (e.g. see #5889), then I think the description should say that.

Reply:
That was the goal of the PR, yes. As it turns out, patching `install_req_from_line` to allow calling it with `comes_from=None` (for easier testing) does fix an issue.

Reply:
Maybe you can add two news entries: one a `.feature` (enhancement) for the message formatting, and another (`.bugfix`) for the fix portion.

Reply:
Anyway, I'll probably make a PR for the patch mentioned here https://github.com/pypa/pip/pull/5788#issuecomment-427596606, so file URLs are better handled on Unix. And once that's merged, another PR with the test from the aforementioned comment and a fix for #5889. And then rebase this PR.